samples/other/file-transfer-s3-to-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/demo/api/ClientApiController.java

```diff
@@ -0,0 +1,137 @@
+ *       Siemens AG - initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.transfer.demo.api;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.eclipse.dataspaceconnector.spi.message.RemoteMessageDispatcherRegistry;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.types.domain.metadata.DataEntry;
+import org.eclipse.dataspaceconnector.spi.types.domain.metadata.QueryRequest;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+@Consumes({ MediaType.APPLICATION_JSON })
+@Produces({ MediaType.APPLICATION_JSON })
+@Path("/transfer")
+public class ClientApiController {
+
+    private final RemoteMessageDispatcherRegistry dispatcherRegistry;
+
+    private final TransferProcessManager processManager;
+
+    private final Monitor monitor;
+
+    public ClientApiController(
+            RemoteMessageDispatcherRegistry dispatcherRegistry,
+            TransferProcessManager processManager,
+            Monitor monitor
+    ) {
+        this.dispatcherRegistry = dispatcherRegistry;
+        this.processManager = processManager;
+        this.monitor = monitor;
+    }
+
+    @GET
+    @Path("artifacts/{connector}")
+    public Response getArtifacts(@PathParam("connector") String connector) {
+        try {
+            var query = QueryRequest.Builder.newInstance()
+                    .connectorAddress(composeConnectorAddress(connector))
+                    .connectorId(connector)
+                    .queryLanguage("dataspaceconnector")
+                    .query("select *")
+                    .protocol("ids-rest").build();
+
+            @SuppressWarnings({ "unchecked", "rawtypes" }) CompletableFuture<List<String>> future
+                    = (CompletableFuture) dispatcherRegistry.send(List.class, query, () -> null);
+
+            var artifacts = future.get();
+            return Response.ok().entity(artifacts).build();
+        } catch (InterruptedException | ExecutionException e) {
+            monitor.severe("Error serving request", e);
+            return Response.serverError().build();
+        }
+    }
+
+    @POST
+    @Path("data/request")
+    public Response initiateDataRequest(Map<String, String> request) {
+        var connector = (String) request.get("connector");
+        var artifact = (String) request.get("artifact");
+
+        var destinationRegion = System.getProperty("destinationRegion");
```

Comment:
Config settings should come from the context so they can be sourced by a configuration extension. Here's an example of how to do this:

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/e9f13a3a10aaa5a91adde99a2a988dc07cdcb571/extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/Oauth2Extension.java#L80

We're still discussing the key formatting but it will likely be something similar to this: 

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/e9f13a3a10aaa5a91adde99a2a988dc07cdcb571/extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/Oauth2Extension.java#L39


We're also trying to keep settings resolution in the extension classes and passing them as config into controllers and other services.

One other thing that is not documented is we are trying to annotate the key constants with @EdcSetting so that later we can have tooling generate configuration documentation:

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/e9f13a3a10aaa5a91adde99a2a988dc07cdcb571/extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/Oauth2Extension.java#L38

Reply:
Hello! 

But _connector_ and _artifact_ are not config settings, these are request parameters. I believe you meant _destinationRegion_ and _destinationBucket_, right?

Reply:
I've updated the PR

Reply:
Yes, sorry my mistake